archunit/src/test/java/com/tngtech/archunit/testutil/assertion/JavaTypeVariableAssertion.java

```diff
@@ -0,0 +1,335 @@
+            this(context + MARKER, "assertion", "", ", ");
+        }
+
+        private DescriptionContext(String context, String description, String currentElement, String joinString) {
+            this.context = context;
+            this.description = description;
+            this.currentElement = currentElement;
+            this.joinString = joinString;
+        }
+
+        public DescriptionContext describe(String part) {
+            return new DescriptionContext(context.replace(MARKER, part + MARKER), description, part, joinString);
+        }
+
+        public DescriptionContext describeUpperBounds() {
+            String newContext = context.replace(MARKER, " extends " + MARKER);
+            return new DescriptionContext(newContext, description, currentElement, " & ");
+        }
+
+        public DescriptionContext describeLowerBounds() {
+            String newContext = context.replace(MARKER, " super " + MARKER);
+            return new DescriptionContext(newContext, description, currentElement, " & ");
+        }
+
+        public DescriptionContext describeElements(int number) {
+            String elementsPlaceHolder = number > 0 ? joinedPlaceHolders(number) : "[]";
+            return new DescriptionContext(context.replace(MARKER, elementsPlaceHolder), description, currentElement, joinString);
+        }
+
+        public DescriptionContext describeElement(int index, int totalSize) {
+            int maxIndex = totalSize - 1;
+            String prefix = index > 0 ? joinedPlaceHolders(index) + joinString : "";
+            String suffix = index < maxIndex ? joinString + joinedPlaceHolders(maxIndex - index) : "";
+            String newContext = context.replace(MARKER, prefix + MARKER + suffix);
+            String newCurrentElement = this.currentElement + "[" + index + "]";
+            return new DescriptionContext(newContext, description, newCurrentElement, joinString);
+        }
+
+        private String joinedPlaceHolders(int number) {
+            return FluentIterable.from(limit(cycle(PLACEHOLDER), number)).join(Joiner.on(joinString));
+        }
+
+        public DescriptionContext step(String description) {
+            return new DescriptionContext(context, description, currentElement, joinString);
+        }
+
+        public DescriptionContext metaInfo() {
+            String newContext = context.replace(MARKER, "{" + MARKER + "}");
+            return new DescriptionContext(newContext, description, currentElement, joinString);
+        }
+
+        public DescriptionContext describeTypeParameters() {
+            String newContext = context.replace(MARKER, "<" + MARKER + ">");
+            String newJoinString = ", ";
+            return new DescriptionContext(newContext, description, currentElement, newJoinString);
+        }
+
+        @Override
+        public String toString() {
+            String currentElementInfix = !currentElement.isEmpty() ? "[" + currentElement + "]" : "";
```

Comment:
That's because we had a different mindset :joy: I always followed "put the 'interesting' thing in the first branch, no matter the condition". But I think you're right, you have to read both branches anyway, might as well optimize the condition to be as simple as possible :+1:

Reply:
Don't worry about these _"I'd flip the `?:`..."_ comments; they're not worth the effort to change!